### PR TITLE
MASP rewards estimation

### PR DESCRIPTION
**Pull request by batconjurer**
## Describe your changes

Adds a cli command to estimate the amount of MASP rewards that will be accumulated by the next epoch.
This is done by applying the latest set of conversions for each asset again. 

## Checklist before merging 
- [ ] If this PR has some consensus breaking changes, I added the corresponding `breaking::` labels
    - This will require 2 reviewers to approve the changes
- [ ] If this PR requires changes to the docs or specs, a corresponding PR is opened in the `namada-docs` repo
    - Relevant PR if applies: 
- [ ] If this PR affects services such as `namada-indexer` or `namada-masp-indexer`, a corresponding PR is opened in that repo
    - Relevant PR if applies:

### CODE REVIEW

File: crates/apps_lib/src/client/rpc.rs [hunk 1]
```diff
@@ -359,6 +360,29 @@ pub async fn query_proposal_by_id<C: Client + Sync>(
...
 
+/// Estimate MASP rewards for next MASP epoch
+pub async fn query_rewards_estimate(
+    context: &impl Namada,
+    args: args::QueryRewardsEstimate,
+) {
+    let mut shielded = context.shielded_mut().await;
+    let _ = shielded.load().await;
+    // Save the update state so that future fetches can be short-circuited
+    let _ = shielded.save().await;
```
Comment (grarco) [inline crates/apps_lib/src/client/rpc.rs:371]: Do we really need t save immediately after load?

Comment (batconjurer) [reply crates/apps_lib/src/client/rpc.rs:371]: Nope, bad copy pasta. Good catch

File: crates/tests/src/integration/masp.rs [hunk 2]
```diff
@@ -1397,6 +1416,27 @@ fn masp_incentives() -> Result<()> {
     assert!(captured.result.is_ok());
     assert!(captured.contains("nam: 0.063"));
 
+    // Assert the rewards estimate is a number higher than the actual rewards
```
Comment (brentstone) [inline crates/tests/src/integration/masp.rs:1419]: what does this comment mean?

**Review by brentstone — APPROVED** [review]
LGTM! Thx for all the tests. Just left a question, and can you look over my very small cleanup commit 16ddaf3340c8d64572117405d09adee9f4fda4c5 before tagging this PR with `merge`?

File: crates/tests/src/integration/masp.rs (hunk 2 above)
Comment (batconjurer) [reply crates/tests/src/integration/masp.rs:1421]: Just a sanity check. The total amount of rewards issued for an asset should increase with each epoch. So estimations should reflect that.

**Comment by batconjurer** [comment]
> LGTM! Thx for all the tests. Just left a question, and can you look over my very small cleanup commit [16ddaf3](https://github.com/anoma/namada/commit/16ddaf3340c8d64572117405d09adee9f4fda4c5) before tagging this PR with `merge`?

Looks good, thanks for fixing those!

**Pull request merged**